### PR TITLE
#862 Enforce immutability of IterableEnvelope through its iterator

### PR DESCRIPTION
**Pull request by Isammoc**
Pull request for #862

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#890](https://github.com/yegor256/cactoos/issues/890) is now in scope, role is `REV`

File: src/test/java/org/cactoos/iterable/IterableEnvelopeTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import org.junit.Test;
+
+/**
+ * Test case for {@link IterableEnvelope}.
+ *
+ * @since 0.34
```
Comment (llorllale) [inline src/test/java/org/cactoos/iterable/IterableEnvelopeTest.java]: @Isammoc we're on `0.35` now

**Comment by llorllale** [comment]
@Isammoc just one minor comment, see above

**Comment by 0crat** [comment]
The job [#890](https://github.com/yegor256/cactoos/issues/890) is now out of scope

**Pull request merged**